cmd/pulsewatcher/pulsewatcher.go

```diff
@@ -65,9 +65,104 @@ func moveBack(reader io.Reader) {
 	fmt.Print(escape(clearDown))
 }
 
+func displayResultsTable(results [][]string, buffer *bytes.Buffer, notReady bool) {
+	table := tablewriter.NewWriter(buffer)
+	table.SetHeader([]string{
+		"URL",
+		"Network State",
+		"Node State",
+		"Pulse Number",
+		"Active List Size",
+		"Working List Size",
+		"Role",
+		"Error",
+	})
+	table.SetBorder(false)
+
+	table.ClearRows()
+	table.ClearFooter()
+
+	moveBack(buffer)
+	buffer.Reset()
+
+	stateString := insolarReady
+	color := tablewriter.FgHiGreenColor
+	if notReady {
+		stateString = insolarNotReady
+		color = tablewriter.FgHiRedColor
+	}
+
+	table.SetFooter([]string{
+		"", "", "", "",
+		"Insolar State", stateString,
+		"Time", time.Now().Format(time.RFC3339),
+	})
+	table.SetFooterColor(
+		tablewriter.Colors{},
+		tablewriter.Colors{},
+		tablewriter.Colors{},
+		tablewriter.Colors{},
+
+		tablewriter.Colors{},
+		tablewriter.Colors{color},
+
+		tablewriter.Colors{},
+		tablewriter.Colors{},
+	)
+
+	table.AppendBulk(results)
+	table.Render()
+	fmt.Print(buffer)
+}
+
+func parseInt64(str string) int64 {
+	res, err := strconv.ParseInt(str, 10, 64)
+	if err != nil {
+		res = -1
+	}
+	return res
+}
+
+func displayResultsJson(results [][]string, buffer *bytes.Buffer, notReady bool) {
+	type DocumentItem struct {
+		URL string
+		NetworkState string
+		NodeState string
+		PulseNumber int64
+		ActiveListSize int64
+		WorkingListSize int64
+		Role string
+		Error string
+	}
+
+	doc := make([]DocumentItem, len(results))
+
+	for i, res := range results {
```

Comment:
use mapping from not named fields too named fields can cause more trouble in future

Reply:
Sorry, but this time I'm not convinced that a map has any advantages over an array in this regard, nor that this change has any business value. This representation was chosen because it's quite natural for the default table output. I suggest to keep it this way for now.